src/stack-error.js

```diff
@@ -10,25 +10,24 @@ export default class StackError extends Error {
       }
 
       const funcs = words.reverse().map((word) => {
-        const func = new Function(
-          'funcs',
-          `
-        if (funcs.length > 0) {
-          const func = funcs[0];
-          func(funcs.slice(1));
-        } else {
-          throw new Error("↓");
-        }
-      `,
-        );
-        Object.defineProperty(func, 'name', { value: word });
-        return func;
+        return {
+          [word](funcs) {
+            if (funcs.length > 0) {
+              const func = funcs[0];
+              func(funcs.slice(1));
+            } else {
+              throw new Error("↓");
+            }
+          }
+        }[word];
       });
 
       const func = funcs[0];
       func(funcs.slice(1));
     } catch (error) {
       Object.setPrototypeOf(error, StackError.prototype);
+      error.name = 'StackError';
+      error.stack = error.stack.split('\n', words.length + 1).join('\n');
```

Comment:
```suggestion
```

Not sure what this is supposed to do? I see no difference without this line.

Reply:
Ah, I see that you updated the OP to say it's for Firefox. It's probably because Firefox doesn't support the `stackTraceLimit` API

Reply:
yes on Node.js but in Firefox i have this stacktrace :
before :
```
StackError: ↓
    I debugger eval code:19
    have debugger eval code:17
    this debugger eval code:17
    at debugger eval code:17
    the debugger eval code:17
    end debugger eval code:17
    StackError debugger eval code:26
    <anonymous> debugger eval code:1
```
after :
```
StackError: ↓
    I debugger eval code:19
    have debugger eval code:17
    this debugger eval code:17
    at debugger eval code:17
    the debugger eval code:17
    end debugger eval code:17
    StackError debugger eval code:26
```

Reply:
Btw it should end with `end`, not StackError if your message is "I have this at the end"

Reply:
yes but it will not work correctly on Node.js for example if I add 2 instead of 1.
A solution would be to check if `StackError` is included at the end of the stacktrace.